### PR TITLE
byobu help page

### PR DESCRIPTION
**Pull request by cat-bro**
byobu help: some scrolling advice plus the contents of the devhero gist (which turns out to be in the byobu github)
https://github.com/cat-bro/admin-training/blob/byobu_md/byobu.md

there may be things to add here - the scrolling info seems to be the most important for day-to-day operations.  If anything pressing comes up in terms of student support and byobu (I hope not) we can add to this during the week.

### CODE REVIEW

**Comment by cat-bro** [comment]
It seems that I have remapped some of my keybindings which is why certain things work for me that would not work for mac users that have not done this.

The scrolling instructions will still work.

I remapped the key bindings on my mac back in 2019 and I'm not sure whether we should advise people to do this because (a) it takes ages and (b) it disables normal F-key functionality (so the user has to type Fn+F1 where they previously typed F1, etc)

**Comment by nsoranzo** [comment]
Tested most keybindings on Linux Ubuntu 20.04 inside GNOME Terminal, they seem to work fine except for "Alt+Up/Down - (PC users) Scroll up/down one page while in scroll mode": PgUp/PgDown seem to be right keys for this, in fact the same "Alt+Up/Down" keybinding is also mentioned later for "Move focus among sessions".

File: byobu.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+GAT virtual machines use byobu.  Byobu has some useful features such as allowing a session to be split into multiple sessions within the same terminal window.  Some terminal behaviour may be different from what you are used to, particularly navigation through scrollback in the terminal.  Some key bindings are listed here including instructions for scrolling in the terminal.
+
+#### Scrolling
+```
+  F7                            Enter scroll mode
+  Enter                         Exit scroll mode
+  Up                            Scroll up one line while in scroll mode
+  Down                          Scroll down one line while in scroll mode
+  Fn+Alt+Up/Down                (Mac users) Scroll up/down one page while in scroll mode
```
Comment (martenson) [inline byobu.md:11]: Simple `PgUp` and `PgDown` work for me already. When using arrows `fn`+`up` and `fn`+`down` work also.

File: byobu.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+More byobu keybindings (from https://github.com/dustinkirkland/byobu/blob/master/usr/share/doc/byobu/help.tmux.txt)
+
+```
+  F1                             * Used by X11 *
+    Shift-F1                     Display this help
+
+  F2                             Create a new window
+    Shift-F2                     Create a horizontal split
+    Ctrl-F2                      Create a vertical split
```
Comment (martenson) [inline byobu.md:23]: vertical split does not work on MacOS for me

File: byobu.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    Ctrl-Shift-F2                Create a new session
```
Comment (martenson) [inline byobu.md:24]: Create a new session does not work on MacOS

File: byobu.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+  F3/F4                          Move focus among windows
+    Alt-Left/Right               Move focus among windows
```
Comment (martenson) [inline byobu.md:27]: Alt-Left/Right does not work on MacOS

File: byobu.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    Alt-Up/Down                  Move focus among sessions
+    Shift-Left/Right/Up/Down     Move focus among splits
+    Shift-F3/F4                  Move focus among splits
+    Ctrl-F3/F4                   Move a split
```
Comment (martenson) [inline byobu.md:31]: Ctrl-F3/F4 does not seem to work on MacOS/iTerm2

File: byobu.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    Ctrl-Shift-F3/F4             Move a window
```
Comment (martenson) [inline byobu.md:32]: Ctrl-Shift-F3/F4 does not work on MacOS/iTerm2 - instead it just moves focus among windows

File: byobu.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    Shift-Alt-Left/Right/Up/Down Resize a split
+
+  F5                             Reload profile, refresh status
+    Alt-F5                       Toggle UTF-8 support, refresh status
+    Shift-F5                     Toggle through status lines
+    Ctrl-F5                      Reconnect ssh/gpg/dbus sockets
+    Ctrl-Shift-F5                Change status bar's color randomly
```
Comment (martenson) [inline byobu.md:39]: Does not work on MacOS/iTerm2 - but I'll live. :)

File: byobu.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+  F6                             Detach session and then logout
```
Comment (martenson) [inline byobu.md:41]: Maybe mention `byobu ls` and then `byobu attach-session -t 1` to reattach

File: byobu.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    Shift-F6                     Detach session and do not logout
+    Alt-F6                       Detach all clients but yourself
+    Ctrl-F6                      Kill split in focus
```
Comment (martenson) [inline byobu.md:44]: Does not work on MacOS/iTerm2

File: byobu.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+  F7                             Enter scrollback history
```
Comment (martenson) [inline byobu.md:46]: Maybe drop this section given we have a separate scrolling section

File: byobu.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+    Alt-PageUp/PageDown          Enter and move through scrollback
+    Shift-F7                     Save history to $BYOBU_RUN_DIR/printscreen
+
+  F8                             Rename the current window
+    Ctrl-F8                      Rename the current session
```
Comment (martenson) [inline byobu.md:51]: Ctrl-F8  does not work on MacOS/iTerm2

File: byobu.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    Shift-F8                     Toggle through split arrangements
+    Alt-Shift-F8                 Restore a split-pane layout
+    Ctrl-Shift-F8                Save the current split-pane layout
+
+  F9                             Launch byobu-config window
+    Ctrl-F9                      Enter command and run in all windows
```
Comment (martenson) [inline byobu.md:57]: Ctrl-F9 does not work on MacOS/iTerm2, it just opens config

File: byobu.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+    Shift-F9                     Enter command and run in all splits
+    Alt-F9                       Toggle sending keyboard input to all splits
```
Comment (martenson) [inline byobu.md:59]: Alt-F9 does not work on MacOS/iTerm2, it just opens config

File: byobu.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+  F10                            * Used by X11 *
+  F11                            * Used by X11 *
+    Alt-F11                      Expand split to a full window
```
Comment (martenson) [inline byobu.md:63]: Alt-F11 does not work on MacOS/iTerm2

File: byobu.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+    Shift-F11                    Zoom into a split, zoom out of a split
+    Ctrl-F11                     Join window into a vertical split
```
Comment (martenson) [inline byobu.md:65]: Ctrl-F11 does not work on MacOS/iTerm2

File: byobu.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+  F12                            Escape sequence
+    Shift-F12                    Toggle on/off Byobu's keybindings
+    Alt-F12                      Toggle on/off Byobu's mouse support
```
Comment (martenson) [inline byobu.md:69]: Alt-F12 does not work on MacOS/iTerm2

File: byobu.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+    Shift-F9                     Enter command and run in all splits
+    Alt-F9                       Toggle sending keyboard input to all splits
+
+  F10                            * Used by X11 *
+  F11                            * Used by X11 *
+    Alt-F11                      Expand split to a full window
+    Shift-F11                    Zoom into a split, zoom out of a split
+    Ctrl-F11                     Join window into a vertical split
+
+  F12                            Escape sequence
```
Comment (martenson) [inline byobu.md:67]: default MacOS/iTerm2 escape sequence seems to be `Ctrl`+`A`

**Comment by martenson** [comment]
> I remapped the key bindings on my mac back in 2019 and I'm not sure whether we should advise people to do this because 

we cannot ask them to do this I think

File: byobu.md (hunk 7 above)
Comment (nsoranzo) [reply byobu.md:39]: We could simplify the list with only the useful keybindings!

File: byobu.md (hunk 17 above)
Comment (hexylena) [reply byobu.md:67]: yeah, default for me too.

File: byobu.md (hunk 16 above)
Comment (hexylena) [reply byobu.md:69]: we can delete this, nothing of ours has mouse support anyway.

File: byobu.md (hunk 5 above)
Comment (hexylena) [reply byobu.md:31]: delete this maybe? not a priority they learn moving splits.

File: byobu.md (hunk 6 above)
Comment (hexylena) [reply byobu.md:32]: same for this + above.

**Comment by hexylena** [comment]
> [...] does not work on MacOS/iTerm2

does anything? :joy: 

same as nicola's response, pgup/dn, everything else is fine (but I don't even use splits ever.)

File: byobu.md (hunk 1 above)
Comment (cat-bro) [reply byobu.md:11]: The mac keyboard I am using and does not have PgUp, PgDown.  I don't think Mac laptop keyboards have PgUp, PgDown keys either.  I can add PgUp/PgDown to this list.

**Comment by cat-bro** [comment]
@hexlyena @martenson @nsoranzo thanks very much for checking this.

Since a lot of things do not work for mac users, I suggest deleting the entire paste of the byobu instructions and replacing it with a link to the devhero byobu gist (same content) with a note that some keybindings will not work for mac users.  There is probably nothing on this page that they will truly need except to know about scroll mode.  (If they follow the ephemeris tutorial exactly, they will split their screens to see the logs.  I can add an extra instruction to the tutorial text for mac users to close the splits with Ctrl+D).

So we would just have (updated) scrolling advice and a link to the gist.

**Comment by hexylena** [comment]
I think we should keep this, it's a great start. Most users will probably ignore it completely, but there will be a couple who try these things and probably contribute back changes to us. So, I think it's in our best interest to keep this and crowdsource the improvements.

**Pull request merged**